tensorflow/compiler/tf2tensorrt/convert/ops/binary_ops.cc

```diff
@@ -70,11 +78,16 @@ class ConvertBinaryImpl {
           "' received both input as constant");
     }
 
-    if (!not_supported_ops.empty() && params.use_implicit_batch) {
-      const auto& end = not_supported_ops.end();
-      if (std::find(not_supported_ops.begin(), end, op) != end) {
-        return errors::Unimplemented(
-            "Binary op: '", op, "' is not supported in implicit batch mode");
+    convertToBool_ = false;
+    if (!not_supported_ops.empty()) {
+      if (find_name(op, not_supported_ops)) {
+        if (params.use_implicit_batch) {
+          return errors::Unimplemented(
+              "Binary op: '", op, "' is not supported in implicit batch mode");
+        } else {
+          // No need to convert the output of "LogicalOr" and "LogicalAnd"
+          convertToBool_ = !both_tensors;
```

Comment:
I think we should just re-name the parameter "both_tensors" to convert_to_bool, so that the implement for ConvertBinary DONOT have to pass both_tensors==false and create confusion.

Reply:
Unfortunately, renaming the `both_tensors` input parameter to `convert_to_bool` will NOT solve the confusion. but, perhaps, create another confusion for the implementation of  `ConvertBooleanBinary::Validate()`. 

The fact is, we need to convert the results to bool **only** for `Greater`, `Less`, `Equal`, `GreaterEqual`, and `LessEqual` binary operators. On the other hand, we need to check if both inputs are tensor only for `LogicalNot` and `LogicalOr` (and, perhaps, `LogicalXor` when it will be implemented).

Therefore,  `convert_to_bool` cannot just replace  `both_tensors`. 
I think the best way to handle this situation 
- reorder the input parameters of 
 ```
ValidateImpl(const OpConverterParams& params,
              const std::vector<string>& implicit_batch_not_supported_ops = {},
              bool both_tensors = false)
 ```
- calculate `convert_to_bool_` internaly (make it `true` when the operator is in `implicit_batch_not_supported_ops` list);
- make it `false` for `LogicalNot` and `LogicalOr` (when `both_tensors` is equal to `true`).

In that case 

- `both_tensors == false` is NOT **directly used** to `ValidateImpl` in `ConvertBinary::Validate()`;
- the variable `convert_to_bool_` is set to `false` under following condition
```
if (both_tensors) {
...
  // No need to convert the output of "LogicalOr" and "LogicalAnd"
  convertToBool_ = false;
}
```